### DISCUSSION TITLE
Minio Primary Api can give data from Minio Secondary when using bucket replication

### DISCUSSION THREAD

**Discussion by triShant-Maharjan** [minio/minio#20702] [Q&A]
I am testing a minio setup for replication where minio A bucket archives is liked to minio B bucket archives. When I upload data to minio B, yes minio B, not A, the data doesn't exist in minio A as replication is from A -> B. But if I use the api of minio A to pull the data using minio client, I can get the data on minio B. I have checked the UI and path, the data is not pulled to minio A. Can anyone explain how this is possible? I have checked the documentation but this usecase is not explained there.

**Comment by klauspost** [comment]
Requests are forwarded to replica if not found locally, so you don't have to wait for your data to eventually be replicated.
